overlord/state/taskrunner.go

```diff
@@ -243,26 +243,17 @@ func (r *TaskRunner) run(t *Task) {
 				t.At(timeNow().Add(x.After))
 			}
 		case nil:
-			var next []*Task
 			switch t.Status() {
 			case DoingStatus:
 				t.SetStatus(DoneStatus)
-				fallthrough
-			case DoneStatus:
-				next = t.HaltTasks()
 			case AbortStatus:
 				// It was actually Done if it got here.
 				t.SetStatus(UndoStatus)
 				r.state.EnsureBefore(0)
 			case UndoingStatus:
 				t.SetStatus(UndoneStatus)
-				fallthrough
-			case UndoneStatus:
-				next = t.WaitTasks()
-			}
-			if len(next) > 0 {
-				r.state.EnsureBefore(0)
 			}
+			r.state.EnsureBefore(0)
```

Comment:
Does this also do the right thing when more than one task had cleanup?

Reply:
as the loop in taskrunner.go calls clean on all relevant tasks it seems it should